### ISSUE TITLE
Foralls not always displaying

### ISSUE THREAD

**Issue by roboguy13** [ku-fpg/hermit#131]
I ran into an issue where there was a forall that wasn't being displayed in a branch of a conjunction resulting from an induction. As a result, it looks like there is an equivalence, but `eqRhs` doesn't work (unless you use `forall-body` first). Running `hermit` on `examples/laws/ListLaws.hs`:

```
hermit<0> rule-to-lemma map-append
map-append (Not Proven)
  ∀ △ △ f x y. map ▲ ▲ f ((++) ▲ x y) ≡ (++) ▲ (map ▲ ▲ f x) (map ▲ ▲ f y)
hermit<1> prove-lemma map-append
Goal:
  ∀ △ △ f x y. map ▲ ▲ f ((++) ▲ x y) ≡ (++) ▲ (map ▲ ▲ f x) (map ▲ ▲ f y)
proof<2> induction 'x
Goal:
  ∀ △ △ f y.
  (map ▲ ▲ f ((++) ▲ (undefined ▲) y) ≡ (++) ▲ (map ▲ ▲ f (undefined ▲)) (map ▲ ▲ f y))
  ∧
  ((map ▲ ▲ f ((++) ▲ ([] ▲) y) ≡ (++) ▲ (map ▲ ▲ f ([] ▲)) (map ▲ ▲ f y))
   ∧
   (∀ a b.
    (map ▲ ▲ f ((++) ▲ b y) ≡ (++) ▲ (map ▲ ▲ f b) (map ▲ ▲ f y))
    ⇒
    (map ▲ ▲ f ((++) ▲ ((:) ▲ a b) y) ≡ (++) ▲ (map ▲ ▲ f ((:) ▲ a b)) (map ▲ ▲ f y))))
proof<3> forall-body
Goal:
  (map ▲ ▲ f ((++) ▲ (undefined ▲) y) ≡ (++) ▲ (map ▲ ▲ f (undefined ▲)) (map ▲ ▲ f y))
  ∧
  ((map ▲ ▲ f ((++) ▲ ([] ▲) y) ≡ (++) ▲ (map ▲ ▲ f ([] ▲)) (map ▲ ▲ f y))
   ∧
   (∀ a b.
    (map ▲ ▲ f ((++) ▲ b y) ≡ (++) ▲ (map ▲ ▲ f b) (map ▲ ▲ f y))
    ⇒
    (map ▲ ▲ f ((++) ▲ ((:) ▲ a b) y) ≡ (++) ▲ (map ▲ ▲ f ((:) ▲ a b)) (map ▲ ▲ f y))))
proof<4> conj-rhs
Goal:
  (map ▲ ▲ f ((++) ▲ ([] ▲) y) ≡ (++) ▲ (map ▲ ▲ f ([] ▲)) (map ▲ ▲ f y))
  ∧
  (∀ a b.
   (map ▲ ▲ f ((++) ▲ b y) ≡ (++) ▲ (map ▲ ▲ f b) (map ▲ ▲ f y))
   ⇒
   (map ▲ ▲ f ((++) ▲ ((:) ▲ a b) y) ≡ (++) ▲ (map ▲ ▲ f ((:) ▲ a b)) (map ▲ ▲ f y)))
proof<5> conj-lhs
Goal:
  map ▲ ▲ f ((++) ▲ ([] ▲) y) ≡ (++) ▲ (map ▲ ▲ f ([] ▲)) (map ▲ ▲ f y)
proof<6> eq-rhs    -- *** This looks like it should work ***
Error in expression: eq-rhs
invalid path.
proof<6> forall-body 
Goal:
  map ▲ ▲ f ((++) ▲ ([] ▲) y) ≡ (++) ▲ (map ▲ ▲ f ([] ▲)) (map ▲ ▲ f y)
proof<7> eq-rhs
Goal:
  (++) ▲ (map ▲ ▲ f ([] ▲)) (map ▲ ▲ f y)
```

**Comment by xich** [comment]
Definitely a bug.

Likely due to some last minute refactoring before my defense... we used to pretty print empty foralls, but that was annoying, so I stopped printing them and tried to make sure we normalize them away when we construct lemmas. This normalization must not be happening somewhere.

I would like to change the `Clause` type so `Forall` only binds a single quantifier, akin to `Lam` in Core. Multiple quantifiers would just be nested foralls. This should simplify the implementation of many of the primitive operations on `Clause`. Maybe that is worth doing instead of fixing this directly.

I should be back online by next week, in case no one does it before then.
